lib/cocoapods/target/aggregate_target.rb

```diff
@@ -292,7 +292,7 @@ def resource_paths_by_config
                 extname = File.extname(resource_path)
                 if self.class.resource_extension_compilable?(extname)
                   output_extname = self.class.output_extension_for_resource(extname)
-                  built_product_dir.join(File.basename(resource_path, extname)).sub_ext(output_extname).to_s
+                  built_product_dir.join(change_ext(File.basename(resource_path), output_extname)).to_s
```

Comment:
@dnkoutso 
This is actually unnecessary. I realize I've used `sub_ext` incorrectly in my impl.  
The correct use would be: 
```
built_product_dir.join(resource_path).sub_ext(output_extname).to_s
```

Reply:
updating!

Reply:
@igor-makarov fixed! the test passes too with your recommendation.

Reply:
actually @igor-makarov this is not correct still:

<img width="1792" alt="Screen_Shot_2020-12-31_at_11_28_34_AM" src="https://user-images.githubusercontent.com/310370/103421778-6267a980-4b5b-11eb-86c7-71e2ea6ca9f8.png">

Reply:
I think we need:

```ruby
                  built_product_dir.join(File.basename(resource_path)).sub_ext(output_extname).to_s
```

Reply:
also updated tests to catch this.

Reply:
Yes! This is the right one!